pkg/flameql/flameql.go

```diff
@@ -83,6 +83,19 @@ func ValidateTagKey(k string) error {
 	return nil
 }
 
+// ValidateTagKeyWithoutReserved report an error if the given tag key violates constraints which NOT include reserved key
+func ValidateTagKeyWithoutReserved(k string) error {
+	if len(k) == 0 {
+		return ErrTagKeyIsRequired
+	}
+	for _, r := range k {
+		if !IsTagKeyRuneAllowed(r) {
+			return newInvalidTagKeyRuneError(k, r)
+		}
+	}
+	return nil
+}
+
```

Comment:
I think we also need to validate the app name. What do you thing about moving this validation to [`segment key`](https://github.com/pyroscope-io/pyroscope/blob/main/pkg/storage/segment/key.go), e.g `segment.ValidateKey(k *Key)`?  We already validate syntax there when parse a string key representation, and a function for validating a structured representation is what the package lacks